components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java

```diff
@@ -688,4 +691,53 @@ private void informRestVersiontoUser(String restVersion) {
+     *
+     */
+    private void extractPlatformAndRuntime() {
+        try {
+            String libPath = GatewayCmdUtils.getCLILibPath();
+            String baloPath = GatewayCliConstants.CLI_GATEWAY + File.separator + GatewayCliConstants.CLI_BALO;
+            String breLibPath = GatewayCliConstants.CLI_BRE + File.separator + GatewayCliConstants.CLI_LIB;
+            String runtimeExtractedPath = libPath + File.separator + GatewayCliConstants.CLI_RUNTIME;
+            String platformExtractedPath =
+                    GatewayCmdUtils.getCLILibPath() + File.separator + GatewayCliConstants.CLI_PLATFORM;
+            try {
+                boolean isChangesDetected = HashUtils.detectChangesInLibraries();
+                // Delete already extracted files if changes detected.
+                if(isChangesDetected) {
+                    Files.deleteIfExists(Paths.get(runtimeExtractedPath));
+                    Files.deleteIfExists(Paths.get(platformExtractedPath));
+                }
+            } catch (HashingException e) {
+                logger.error("Error while detecting changes in gateway libraries", e);
+            }
+            if (!Files.exists(Paths.get(runtimeExtractedPath))) {
+                ZipUtils.unzip(runtimeExtractedPath + GatewayCliConstants.EXTENSION_ZIP, runtimeExtractedPath, false);
+                //copy balo to the runtime
+                GatewayCmdUtils.copyFolder(libPath + File.separator + baloPath,
+                        runtimeExtractedPath + File.separator + GatewayCliConstants.CLI_LIB + File.separator
+                                + GatewayCliConstants.CLI_REPO);
+                //copy gateway jars to runtime
+                GatewayCmdUtils.copyFolder(libPath + File.separator + GatewayCliConstants.CLI_GATEWAY + File.separator
+                        + GatewayCliConstants.CLI_RUNTIME, runtimeExtractedPath + File.separator + breLibPath);
+            }
+            if (!Files.exists(Paths.get(platformExtractedPath))) {
+                ZipUtils.unzip(platformExtractedPath + GatewayCliConstants.EXTENSION_ZIP, platformExtractedPath, true);
+                //copy balo to the platform
+                GatewayCmdUtils.copyFolder(libPath + File.separator + baloPath,
+                        platformExtractedPath + File.separator + GatewayCliConstants.CLI_LIB + File.separator
+                                + GatewayCliConstants.CLI_REPO);
+                //copy gateway jars to platform
+                GatewayCmdUtils.copyFolder(libPath + File.separator + GatewayCliConstants.CLI_GATEWAY + File.separator
+                        + GatewayCliConstants.CLI_PLATFORM, platformExtractedPath + File.separator + breLibPath);
+            }
+        } catch (IOException e) {
+            String message = "Error while unzipping platform and runtime while project setup";
+            logger.error(message, e);
+            throw new CLIInternalException(message);
```

Comment:
```suggestion
            throw new CLIInternalException(message, e);
```
Will this be enough without the `logger.error(message, e)`?

Reply:
This will only outputs to standard error, and won't be captured in the log file